lib/dry/web/settings.rb

```diff
@@ -27,16 +26,39 @@ def self.check_schema_duplication(new_schema)
       end
       private_class_method :check_schema_duplication
 
+      def self.extract_environment_data(root, env)
+        env_file = find_env_file(root, env)
+        raise MissingEnvFile, "Try to load #{env} file, but no file found" unless env_file
+        parse_file(env_file)
+      end
+      private_class_method :extract_environment_data
+
+      def self.find_env_file(root, env)
+        root.each_child.find { |path| path.basename.fnmatch(".env.#{env}") }
+      end
+      private_class_method :find_env_file
+
+      def self.parse_file(file)
+        result = {}
+        File.readlines(file).each do |line|
+          key, value = line.split('=')
+          if key && value
+            result[key] = value.strip.sub(/\A(['"])(.*)\1\z/, '\2')
+          end
+        end
+        result
+      end
+      private_class_method :parse_file
```

Comment:
This seems a little simplistic right now (e.g. it won't work as expected if a `=` character appears in one of the values). Might be useful to see how dotenv handles the loading of the files?